cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt

```diff
@@ -187,16 +187,38 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
                 is Python.AST.And -> "and"
                 is Python.AST.Or -> "or"
             }
-        val ret = newBinaryOperator(operatorCode = op, rawNode = node)
-        if (node.values.size != 2) {
-            return newProblemExpression(
+        return handleBoolOpRecursively(node.values, op, node)
```

Comment:
Please consider merging #1700

Reply:
Agree

Reply:
Removed the recursive version with a foldr

Reply:
Merge whichever version you prefer. I don't like this `foldRight` solution because it requires a lot more mental effort to understand what's going on. I think the simple loop is much easier to maintain. However, since this is a matter of taste (and your solution is probably the "Kotlin way" to go) -> I won't block it.